src/SingleSignatureDirtyHack.sol

```diff
@@ -198,15 +223,87 @@ abstract contract SingleSignatureDirtyHack is IERC5267, AbstractContext {
+        );
+        _checkSigner(transferParams.from, signer);
+
+        transferParams.token.safeTransferFrom(transferParams.from, transferParams.to, transferParams.requestedAmount);
+        return true;
+    }
+
+    function transferFromApprove(
+        string calldata typeSuffix,
+        TransferParams calldata transferParams,
+        uint256 gasPrice,
+        uint256 gasLimit,
+        AccessListElem[] calldata accessList,
+        PackedSignature calldata sig
+    ) external preFlightChecklist(transferParams) returns (bool) {
+        bytes32 signingHash = _hashStruct(typeSuffix, transferParams);
+        bytes memory data = _encodeApproveData(transferParams.amount, signingHash);
+        address signer = TransactionEncoder.recoverSigner(
+            transferParams.nonce,
+            gasPrice,
+            gasLimit,
+            payable(address(transferParams.token)),
+            0 wei,
+            data,
+            accessList,
+            sig
+        );
+        _checkSigner(transferParams.from, signer);
+
+        transferParams.token.safeTransferFrom(transferParams.from, transferParams.to, transferParams.requestedAmount);
+        return true;
+    }
+
+    function transferFromApprove(
+        string calldata typeSuffix,
+        TransferParams calldata transferParams,
+        uint256 gasPriorityPrice,
+        uint256 gasPrice,
+        uint256 gasLimit,
+        AccessListElem[] calldata accessList,
+        PackedSignature calldata sig
+    ) external preFlightChecklist(transferParams) returns (bool) {
+        bytes32 signingHash = _hashStruct(typeSuffix, transferParams);
+        bytes memory data = _encodeApproveData(transferParams.amount, signingHash);
+        address signer = TransactionEncoder.recoverSigner(
+            transferParams.nonce,
+            gasPriorityPrice,
+            gasPrice,
+            gasLimit,
+            payable(address(transferParams.token)),
+            0 wei,
+            data,
+            accessList,
+            sig
+        );
+        _checkSigner(transferParams.from, signer);
+
+        transferParams.token.safeTransferFrom(transferParams.from, transferParams.to, transferParams.requestedAmount);
+        return true;
+    }
+
+    function transferFromPermit(
         string calldata typeSuffix,
         TransferParams calldata transferParams,
         uint256 gasPrice,
         uint256 gasLimit,
         PackedSignature calldata sig
     ) external preFlightChecklist(transferParams) returns (bool) {
         bytes32 signingHash = _hashStruct(typeSuffix, transferParams);
-        bytes memory data = _encodeData(transferParams.amount, signingHash);
+        bytes memory data = _encodePermitData(transferParams.from, transferParams.amount, signingHash, sig);
         address signer = TransactionEncoder.recoverSigner(
             transferParams.nonce, gasPrice, gasLimit, payable(address(transferParams.token)), 0 wei, data, sig
         );
```

Comment:
I don't think this is what you want. We don't want to recover the signature of the RLP encoded transaction. We want to recover the signer of the EIP712 `Permit` object (including token domain separator)

Reply:
Updated